pinot-controller/src/main/java/com/linkedin/pinot/controller/util/TableSizeReader.java

```diff
@@ -199,35 +208,41 @@ public TableSubTypeSizeDetails getTableSubtypeSize(String tableNameWithType, int
               segmentEntry.getKey(), errors, segmentLevelMax);
         }
         segmentSizes.estimatedSizeInBytes = segmentSizes.reportedSizeInBytes + errors * segmentLevelMax;
-        tableLevelMax = Math.max(tableLevelMax, segmentLevelMax);
         subTypeSizeDetails.reportedSizeInBytes += segmentSizes.reportedSizeInBytes;
         subTypeSizeDetails.estimatedSizeInBytes += segmentSizes.estimatedSizeInBytes;
       } else {
         LOGGER.warn("Could not get size report from any server for segment {}", segmentEntry.getKey());
         segmentSizes.reportedSizeInBytes = -1;
         segmentSizes.estimatedSizeInBytes = -1;
+        subTypeSizeDetails.missingSegments++;
       }
     }
-    if (tableLevelMax == -1) {
-      LOGGER.warn("Could not get size from any of the servers for table {}. Setting size estimate to -1",
-          tableNameWithType);
-      // no server reported size
-      subTypeSizeDetails.reportedSizeInBytes = -1;
-      subTypeSizeDetails.estimatedSizeInBytes = -1;
-    } else {
-      LOGGER.info("Table {} max segment size is {}", tableNameWithType, tableLevelMax);
-      // For segments with no reported sizes, use max table-level segment size as an estimate
-      for (Map.Entry<String, SegmentSizeDetails> segmentSizeDetailsEntry : segmentMap.entrySet()) {
-        SegmentSizeDetails sizeDetails = segmentSizeDetailsEntry.getValue();
-        if (sizeDetails.reportedSizeInBytes != -1) {
-          continue;
-        }
-        LOGGER.info("Using tableLevelMax {} for segment {} size estimation", tableLevelMax, segmentSizeDetailsEntry.getKey());
-        sizeDetails.estimatedSizeInBytes += sizeDetails.serverInfo.size() * tableLevelMax;
-        subTypeSizeDetails.estimatedSizeInBytes += sizeDetails.estimatedSizeInBytes;
+    if (subTypeSizeDetails.missingSegments > 0) {
+      if (subTypeSizeDetails.missingSegments == subTypeSizeDetails.segments.size()) {
+        LOGGER.warn("Could not get size reports from any server for all segments of the table {}", table);
+        subTypeSizeDetails.reportedSizeInBytes = -1;
+        subTypeSizeDetails.estimatedSizeInBytes = -1;
+        _controllerMetrics.setValueOfTableGauge(table, ControllerGauge.TABLE_STORAGE_ESTIMATION_FAILURE,
+            1);
+        // reset metric tracking partial estimation
+        _controllerMetrics.setValueOfTableGauge(table, ControllerGauge.TABLE_STORAGE_EST_FAILED_SEGMENTS,
```

Comment:
shouldn't this metric be set to segments.size()? we failed to get size for all segments, right? And then maybe we don't need the other metric at all?